controllers/users.controller.js

```diff
@@ -155,4 +160,58 @@ const updateAvatars = async (req, res) => {
   });
 };
 
-export default { signupUser, loginUser, logoutUser, currentUser, updateAvatars };
+const verifyEmail = async (req, res) => {
+  try {
+    const { verificationToken } = req.params;
+    const user = await User.findOne({ verificationToken });
+    if (!user) {
+      return res.status(404).json({ message: 'User not found' });
+    }
+    user.set('verify', true);
+    user.verificationToken = null;
+    await user.save();
+    return res.status(200).json({ message: 'Verification successful' });
+  } catch (error) {
+    console.error(error);
+  }
+};
+
+const reverifyEmail = async (req, res) => {
+  try {
+    const { email } = req.body;
+    if (!email) {
+      return res.status(400).json({ message: 'Missing required field email' });
+    }
+    const user = await User.findOne({ email });
+    if (user.verify) {
+      return res.status(400).json({
+        status: 'Bad Request',
+        code: 400,
+        ResponseBody: {
+          message: 'Verification has already been passed',
+        },
+      });
+    }
+    const verificationToken = user.verificationToken;
+    send(email, verificationToken);
+    res.status(200).json({
+      status: 'Ok',
+      code: 200,
+      ResponseBody: {
+        message: 'Verification email sent',
+      },
+    });
+  } catch (error) {
+    console.error(error);
+  }
+};
+
+export default {
+  signupUser,
```

Comment:
Posortowałbym to aby łatwiej było potem szukać. Ale nie tylko te exporty ale ogólnie importy też warto sortować alfabetycznie.